### PR TITLE
Remove unused regions

### PR DESCRIPTION
**Pull request by ThetaSinner**
### Summary

Follow up from https://github.com/holochain/holochain/pull/2656#discussion_r1303305708

### TODO:
- [x] https://github.com/holochain/holochain/pull/2762
- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

**Comment by ThetaSinner** [comment]
Adding backport label because we're making changes in this area so it should make other backports simpler to backport this too

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net_task.rs [hunk 1]
```diff
@@ -1880,6 +2343,42 @@ mod tests {
...
+        tokio::time::timeout(Duration::from_millis(1000), async {
+            while !meta_net_task_finished.load(Ordering::Acquire) {
+                tokio::time::sleep(Duration::from_millis(1)).await;
+            }
+        })
+        .await
+        .expect("Timed out waiting for shutdown");
+
+        assert!(meta_net_task_finished.load(Ordering::Acquire));
+    }
```
Comment (maackle) [inline crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net_task.rs:2380]: If this took a closure, you could clean up the other instances using this same pattern

```suggestion
    async fn wait_and_assert_shutdown(finished: impl Fn() -> bool) {
        tokio::time::timeout(Duration::from_millis(1000), async {
            while !finished() {
                tokio::time::sleep(Duration::from_millis(1)).await;
            }
        })
        .await
        .expect("Timed out waiting for shutdown");

        assert!(finished());
    }
```

**Review by maackle — APPROVED** [review]
This is a lot more than removing the Regions variant, did you mean to base it on some other branch? Either way it all looks good.

**Comment by ThetaSinner** [comment]
> This is a lot more than removing the Regions variant, did you mean to base it on some other branch? Either way it all looks good.

This PR is based on the PR linked in the description. I meant to leave this one in draft for the moment! I'll address your comments when the other PR goes in

**Review by neonphog — APPROVED** [review]

**Pull request merged**